migrations/20230201519590_add_oauth_state_table.up.sql

```diff
@@ -0,0 +1,14 @@
+-- auth.oauth_state definition
+create table if not exists {{ index .Options "Namespace" }}.oauth_state(
+       id uuid primary key,
+       user_id uuid null,
+       supabase_auth_code text unique not null,
```

Comment:
Couple of things regarding this column:

1. Can we avoid the use of `supabase` prefixes? Why not just use `auth_code`?
2. Can we avoid using `unique` unless absolutely necessary. When the auth code is a UUID, it's practically impossible to get a duplicate one. The reason I'm no longer a fan of `unique` is that users may decide to reference this column and then we won't be able to change it. 🙃 We do need a basic index for the lookup though.